### ISSUE TITLE
add_intracellular_recording raises ValueError: row data keys don't match available columns

### ISSUE THREAD

**Issue by lvsltz** [oruebel/ndx-icephys-meta#15]
After the last updates, I am not able to `add_intracellular_recording`s anymore:
For example in https://github.com/oruebel/ndx-icephys-meta/blob/master/src/pynwb/examples/icephys_meta_simple_example.ipynb
```
nwbfile.add_intracellular_recording(electrode=electrode, stimulus, response, 10)
```
raises a ValueError. Full error is pasted at the end.
```
ValueError: row data keys don't match available columns
you supplied 1 extra keys: {'enforce_unique_id'}
and were missing 0 keys: set()
```

Is any of these packages at the wrong version?
```
h5py                      2.10.0                   pypi_0    pypi
hdmf                      1.3.3                    pypi_0    pypi
ndx-icephys-meta          0.1.0                     dev_0    <develop>
numpy                     1.17.3                   pypi_0    pypi
nwb-docutils              0.3.1                    pypi_0    pypi
pandas                    0.25.2                   pypi_0    pypi
pynwb                     1.1.2                    pypi_0    pypi
python                    3.7.4                h265db76_1  
ruamel-yaml               0.16.5                   pypi_0    pypi
ruamel-yaml-clib          0.2.0                    pypi_0    pypi 
```

```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-5-6537f1a1d799> in <module>
      2                                     stimulus=stimulus,
      3                                     response=response,
----> 4                                     id=10)

~/miniconda3/envs/ndx/lib/python3.7/site-packages/hdmf/utils.py in func_call(*args, **kwargs)
    436                         raise_from(ExceptionType(msg), None)
    437 
--> 438                 return func(self, **parsed['args'])
    439         else:
    440             def func_call(*args, **kwargs):

/nfs4/bbp.epfl.ch/user/username/private/git/ndx-icephys-meta/src/pynwb/ndx_icephys_meta/icephys.py in add_intracellular_recording(self, **kwargs)
    482         self._check_intracellular_recordings()
    483         # Add the recoding to the intracellular_recordings table
--> 484         call_docval_func(self.intracellular_recordings.add_recording, kwargs)
    485 
    486     def _check_ic_sweeps(self):

~/miniconda3/envs/ndx/lib/python3.7/site-packages/hdmf/utils.py in call_docval_func(func, kwargs)
    325 def call_docval_func(func, kwargs):
    326     fargs, fkwargs = fmt_docval_args(func, kwargs)
--> 327     return func(*fargs, **fkwargs)
    328 
    329 

~/miniconda3/envs/ndx/lib/python3.7/site-packages/hdmf/utils.py in func_call(*args, **kwargs)
    436                         raise_from(ExceptionType(msg), None)
    437 
--> 438                 return func(self, **parsed['args'])
    439         else:
    440             def func_call(*args, **kwargs):

/nfs4/bbp.epfl.ch/user/username/private/git/ndx-icephys-meta/src/pynwb/ndx_icephys_meta/icephys.py in add_recording(self, **kwargs)
    114                       'response': (response_start_index, response_index_count, response)}
    115         row_kwargs.update(kwargs)
--> 116         return super(IntracellularRecordings, self).add_row(enforce_unique_id=True, **row_kwargs)
    117 
    118 

~/miniconda3/envs/ndx/lib/python3.7/site-packages/hdmf/utils.py in func_call(*args, **kwargs)
    436                         raise_from(ExceptionType(msg), None)
    437 
--> 438                 return func(self, **parsed['args'])
    439         else:
    440             def func_call(*args, **kwargs):

~/miniconda3/envs/ndx/lib/python3.7/site-packages/hdmf/common/table.py in add_row(self, **kwargs)
    325                     'row data keys don\'t match available columns',
    326                     'you supplied {} extra keys: {}'.format(len(extra_columns), extra_columns),
--> 327                     'and were missing {} keys: {}'.format(len(missing_columns), missing_columns)
    328                 ])
    329             )

ValueError: row data keys don't match available columns
you supplied 1 extra keys: {'enforce_unique_id'}
and were missing 0 keys: set()
```

**Comment by oruebel** [comment]
This should be fixed on https://github.com/oruebel/ndx-icephys-meta/pull/13  This PR requires 
https://github.com/hdmf-dev/hdmf/pull/191 We need to do a release of HDMF and PyNWB in order merge that PR, but I expect that we should be able to do those today.

**Comment by lvsltz** [comment]
Thanks for the quick reply! I had managed to get past this error with a `pip install git+` on the last commit of that PR (https://github.com/hdmf-dev/hdmf/pull/191) but I see you merged it meanwhile.
Now I get other issues from hdmf.
Please let me know if there's a commit that would allow me to fully use the extension.

I'd like to be able to test it these days, before our call on Thursday.

**Comment by oruebel** [comment]
Will do. We are going through PRs on HDMF right now. I expect that we'll have this all running again later today.

**Comment by oruebel** [comment]
@lvsltz please try again with the current dev branch of PyNWB and the latest HDMF release. We have now:

1. released HDMF 1.4.0
1. updated the dev branch of PyNWB to work with HDMF 1.4.0, and 
1. merged #13 

I believe these changes should fix the issues you have been seeing and all the example notebooks in this repo should be working now. Please reopen this issue or create a new one in case the issue should persist or you see new issues.
